.github/workflows/firebase-hosting-production.yml

```diff
@@ -1,25 +1,59 @@
 # This file was auto-generated by the Firebase CLI
 # https://github.com/firebase/firebase-tools
 
-name: Deploy to Firebase Hosting (production)
+name: Deploy to Firebase Hosting (production) with Pre-Release Tests
 'on':
   push:
     tags:
       - 'v[0-9]+.[0-9]+.[0-9]+'
       - 'v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+'
       - 'v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+'
 jobs:
-  build_and_deploy:
-    name: Deploy
+  pre-release-tests:
+    name: Run Cypress Pre-Release Tests
+    needs: [build-and-preview]
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    strategy:
+      fail-fast: false
+      matrix:
+        browser: [chromium, edge]
+        containers: [1, 2, 3, 4]
+    env:
+      NODE_ENV: 'test'
+      CYPRESS_BASE_URL: 'http://localhost:5173'
```

Comment:
Should this use the staging link instead of localhost?

Reply:
I don't think so, because we are testing the open administrations that have been synced from production Firestore to development Firestore. Running on dev will also allow us to run super admin tests which can clean up the dev database on a new relese (i.e. deleting all test runs).